docs/components/merak_network.md

```diff
@@ -0,0 +1,118 @@
+# Merak Network Design Document  
+
+- [Merak Network Design Document](#merak-network-design-document)
+  - [Overview](#overview)
+  - [Service Requirements](#service-requirements)
+  - [Design](#design)
+  - [Data Schema](#data-schema)
+
+## Overview  
+___  
+The Merak Network component is serving the purpose of creating modifying and deleting the Network configuration from the Scenario Manager. In our first version, it will support configuring network configurations with Alcor.  
+
+## Service Requirements  
+___  
+1. Ability to create/modify/delete network configurations in Alcor  
+2. Create Security Groupe  
+3. Create/modify/delete Number of VPC 
+4. Create/modify/delete Number of Subnet per VPC  
+5. Create Router  
+6. Connect Subnets to Router  
+7. node registration
+8. Store the above information in DB, and have the ability to delete them after each test run. Or have ability to modify our network configurations on the fly.  
+
+## Design  
+___  
+
+- As for the first version of Merak Network component, it will mainly setup virtual network with [Alcor](https://github.com/futurewei-cloud/alcor).  
+- To enable Merak Network have the ability to work with different SDN platform, such as [Alcor](https://github.com/futurewei-cloud/alcor). For now we will have different functions within the code to do different SDN network setup, later may switch to use different micro service do different SDN.  
+- Once received request from the Merak Scenario Manager, Merak Network will start to do its job. Making VPC, Security Group, Router, Subnet; and attach them together.  
+- After setup the virtual network, all information related to future use will be saved into DB. Currently planning on using the mongoDB, the schema will be show in the below section.  
+
+___  
+
+Merak Network will have couple of functions:  
```

Comment:
@yanmo96 Can you layout the workflow for these functions? You can use a block to represent a process and draw a line between processes to show the workflow. You can refer to Zhen's design.

Reply:
Hi @cj-chung is the following graph ok? Thanks
https://raw.githubusercontent.com/yanmo96/merak-1/network_doc/docs/images/MerakNetwork_v2.drawio.svg

Reply:
ok

Reply:
Hi @cj-chung, I have update the image and DB according to our meeting earlier. Please let me know if is ok, Thanks